shadowsocks/tcp.go

```diff
@@ -226,7 +241,7 @@ func (s *tcpService) handleConnection(listenerPort int, clientConn onet.DuplexCo
 
 	connStart := time.Now()
 	clientConn.(*net.TCPConn).SetKeepAlive(true)
-	// Set a deadline for connection authentication
+	// Set a deadline to establish a connection to the target.
```

Comment:
I don't think this is accurate.  We clear the deadline before attempting the connection to the target.